indra/newview/lltoolpie.cpp

```diff
@@ -1572,8 +1572,11 @@ bool LLToolPie::shouldAllowFirstMediaInteraction(const LLPickInfo& pick, bool mo
     // Check if the object is owned by a friend of the agent
     if(FirstClickPref & MEDIA_FIRST_CLICK_FRIEND)
     {
-        LL_DEBUGS_ONCE() << "FirstClickPref & MEDIA_FIRST_CLICK_FRIEND. id: " << owner_id << LL_ENDL;
-        return LLAvatarTracker::instance().isBuddy(owner_id);
+        if(LLAvatarTracker::instance().isBuddy(owner_id))
+        {
+            LL_DEBUGS_ONCE() << "FirstClickPref & MEDIA_FIRST_CLICK_FRIEND. id: " << owner_id << LL_ENDL;
```

Comment:
'debugs' and 'infos' options are the most unsefull when they have a tag, so that instead of having to toggle 'debug' for multiple files individually, we can use log control to, as an example, turn on "MediaFirstClick' only and it will start working across multiple files.